task-driver/src/create_new_wallet.rs

```diff
@@ -207,78 +206,60 @@ impl NewWalletTask {
-    async fn generate_proof(&self) -> Result<ValidWalletCreateBundle, NewWalletTaskError> {
+    async fn generate_proof(&mut self) -> Result<(), NewWalletTaskError> {
         // Index the wallet in the global state
         self.global_state.add_wallets(vec![self.wallet.clone()]).await;
 
         // Construct the witness and statement for the proof
-        let witness =
-            ValidWalletCreateWitness { private_wallet_share: self.wallet.private_shares.clone() };
-
-        let private_shares_commitment =
-            compute_wallet_private_share_commitment(&self.wallet.private_shares);
-        let statement = ValidWalletCreateStatement {
-            private_shares_commitment,
-            public_wallet_shares: self.wallet.blinded_public_shares.clone(),
-        };
+        let (witness, statement) = self.get_witness_statement();
 
         // Enqueue a job with the proof manager to prove `VALID NEW WALLET`
         let (response_sender, response_receiver) = oneshot::channel();
         let job_req = ProofManagerJob {
             type_: ProofJob::ValidWalletCreate { statement, witness },
             response_channel: response_sender,
         };
+
         self.proof_manager_work_queue
             .send(job_req)
             .map_err(|err| NewWalletTaskError::SendMessage(err.to_string()))?;
 
+        // Await the proof
         let proof_bundle = response_receiver
             .await
             .map_err(|err| NewWalletTaskError::ProofGeneration(err.to_string()))?;
-
-        Ok(proof_bundle.into())
+        self.proof = Some(proof_bundle.into());
+        Ok(())
     }
 
     /// Submit the newly created wallet on-chain with proof of validity
     async fn submit_wallet_tx(&mut self) -> Result<(), NewWalletTaskError> {
-        let proof = if let NewWalletTaskState::SubmittingTx { proof_bundle } = self.state() {
-            proof_bundle
-        } else {
-            unreachable!("can only begin submitting wallet after proof is generated")
-        };
-
-        // Compute a commitment to the private shares to use on-chain
-        let private_share_commitment = self.wallet.get_private_share_commitment();
-        let tx_hash = self
-            .starknet_client
-            .new_wallet(private_share_commitment, self.wallet.blinded_public_shares.clone(), proof)
-            .await
-            .map_err(|err| NewWalletTaskError::Starknet(err.to_string()))?;
-
-        log::info!("tx hash: 0x{:x}", starknet_felt_to_biguint(&tx_hash));
-        let status = self
-            .starknet_client
-            .poll_transaction_completed(tx_hash)
+        let proof = self.proof.take().unwrap();
+        self.arbitrum_client
+            .new_wallet(proof)
             .await
-            .map_err(|err| NewWalletTaskError::Starknet(err.to_string()))?;
-
-        status.into_result().map_err(|err| NewWalletTaskError::Starknet(err.to_string()))
+            .map_err(|err| NewWalletTaskError::Starknet(err.to_string()))
```

Comment:
confirming that we're going to refactor this error variant to `NewWalletTaskError::Arbitrum` in a future PR?

Reply:
oops forgot to change this, good catch